src/t_zset.c

```diff
@@ -3820,11 +3820,15 @@ void genericZpopCommand(client *c, robj **keyv, int keyc, int where, int emitkey
     }
 
     void *arraylen_ptr = addReplyDeferredLen(c);
-    long arraylen = 0;
+    long result_count = 0;
 
     /* We emit the key only for the blocking variant. */
     if (emitkey) addReplyBulk(c,key);
 
+    /* Respond with a single array in RESP2 and nested arrays in RESP3.
+     * Even in RESP3, use a single array when blocking or when no 'count' argument.  */
+    int use_nested_array = (c->resp > 2 && emitkey == 0 && countarg != NULL);
```

Comment:
I think we don't need to mention the blocking varient, currently it doesn't support the COUNT feature, and when it will, we'd probably like to use nested arrays in some way.
```suggestion
    /* Respond with a single (flat) array in RESP2 or if countarg is not
     * provided (returning a single element). In RESP3, When countarg is
     * provided, use nested array. */
    int use_nested_array = c->resp > 2 && countarg != NULL;
```

Reply:
"Returning a single element" may be misleading. It is still possible to return a nested array even when a single element is returned, in two circumstances: countarg is provided but count=1, or count>1 but there is only one element left to return. Up to you whether you think that needs to be clarified in the comment.

Reply:
it seems clear to me, but feel free to improve.

i.e the condition was that countarg is provided (even if it's 1, or the set has only one item).
the site note (in parenthesis is that in this case it returns just one element).

again, feel free to improve.